cloud/linode/fake_linode_test.go

```diff
@@ -240,6 +278,37 @@ func (f *fakeAPI) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 				_, _ = w.Write(rr)
 				return
 			}
+		case "networking":
+			rx, _ := regexp.Compile("/networking/firewalls/[0-9]+/devices")
+			if rx.MatchString(urlPath) {
+				fwdId, err := strconv.Atoi(strings.Split(urlPath, "/")[3])
+				if err != nil {
+					f.t.Fatal(err)
+				}
+
+				firewallDevices, found := f.fwd[fwdId]
+				if found {
+					firewallDeviceList := []linodego.FirewallDevice{}
+					for i := range firewallDevices {
+						firewallDeviceList = append(firewallDeviceList, *firewallDevices[i])
+					}
+					rr, _ := json.Marshal(linodego.FirewallDevicesPagedResponse{
+						PageOptions: &linodego.PageOptions{Page: 1, Pages: 1, Results: len(firewallDeviceList)},
+						Data:        firewallDeviceList,
+					})
+					_, _ = w.Write(rr)
+				} else {
+					w.WriteHeader(404)
+					resp := linodego.APIError{
+						Errors: []linodego.APIErrorReason{
+							{Reason: "Not Found"},
+						},
+					}
+					rr, _ := json.Marshal(resp)
```

Comment:
It's kinda weird that we check errors from `strconv.Atoi` and not these ones. They seem to be similarly likely to happen.